### PR TITLE
VIH-6195 check audio file in function

### PR DESCRIPTION
**Pull request by valentinabennett**
WIP

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-6195

### Change description ###
Added endpoint to get all closed conferences expired 14hours ago with audio recording

**Does this PR introduce a breaking change?** (check one with "x")

```
[X ] Yes
[ ] No
```

### CODE REVIEW

File: VideoAPI/VideoApi.DAL/Queries/GetExpiredAudiorecordingClosedConferencesQuery.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            _context = context;
+        }
+
+        public async Task<List<Conference>> Handle(GetExpiredAudiorecordingClosedConferencesQuery query)
+        {
+            return await _context.Conferences
+                .Where(x => x.ScheduledDateTime.AddHours(14) <= DateTime.UtcNow
+                && x.AudioRecordingRequired
+                && x.State == ConferenceState.Closed)
```
Comment (RuwanikaWeerakkody) [inline VideoAPI/VideoApi.DAL/Queries/GetExpiredAudiorecordingClosedConferencesQuery.cs:30]: This should be status >0 not == closed. Please see the criteria in the story

File: VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature [hunk 2]
```diff
@@ -3,91 +3,101 @@ Feature: Conference
...
-    And the judge status is in hearing
-    And I have a get judges in hearings today
-    When I send the request to the endpoint
-    Then the response should have the status OK and success status True
-    And the Judges in hearings should be retrieved
+Scenario: Get audiorecording expired closed conferences
+	Given I have a conference with audiorecording
+	And I have another conference with audiorecording
+	And I have a conference for tomorrow with audiorecording
+	And I close all conferences
```
Comment (RuwanikaWeerakkody) [inline VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature:90]: Please update here to match the criteria of status >0

File: VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs [hunk 3]
```diff
@@ -63,7 +63,12 @@ public void GivenIHaveAConference()
             CreateConference(DateTime.Now.ToLocalTime().AddMinutes(2));
         }
 
-        
+        [Given(@"I have a conference with audiorecording")]
+        public void GivenIHaveAConferenceWithAudiorecording()
+        {
+            CreateConference(DateTime.Now.ToLocalTime().AddHours(-14).AddMinutes(-2),null, true);
```
Comment (npoxon) [inline VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs:69]: Should this be UTC time not local time?

File: VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs [hunk 4]
```diff
@@ -78,6 +83,14 @@ public void GivenIHaveAnotherConference()
             _context.Test.ConferenceIds.Add(_context.Test.ConferenceResponse.Id);
         }
 
+
+        [Given(@"I have another conference with audiorecording")]
+        public void GivenIHaveAnotherConferenceWithAudiorecording()
+        {
+            CreateConference(DateTime.Now, null, true);
```
Comment (npoxon) [inline VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs:90]: Same here? UtcNow

File: VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs [hunk 5]
```diff
@@ -14,5 +14,10 @@ public class ExpiredConferencesResponse
         /// The current conference status
         /// </summary>
         public ConferenceState CurrentStatus { get; set; }
+
+        /// <summary>
+        /// The hearing Id
+        /// </summary>
+        public Guid HearingId { get; set; }
```
Comment (npoxon) [inline VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs:21]: on video api I think the references to HearingId are HearingRefId

File: VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetExpiredAudiorecordingClosedConferencesQueryTests.cs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            _conference4Id = conference4.Id;
+
+            await TestDataManager.SeedConference(conference1);
+            await TestDataManager.SeedConference(conference2);
+            await TestDataManager.SeedConference(conference3);
+            await TestDataManager.SeedConference(conference4);
+
+            var conferences = await _handler.Handle(new GetExpiredAudiorecordingClosedConferencesQuery());
+            var confIds = conferences.Select(x => x.Id).ToList();
+            confIds.Should().Contain(conference1.Id);
```
Comment (npoxon) [inline VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetExpiredAudiorecordingClosedConferencesQueryTests.cs:87]: you are not actually asserting that the conferences that have audio recording are retrieved? You are getting a response with a list of conferences, getting their ids, then checking that those conferences contain those ids which will always be true?

File: VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs (hunk 5 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs:21]: I think it's should not match Conference type, it just meaning that we need hearingId in return

File: VideoAPI/VideoApi.DAL/Queries/GetExpiredAudiorecordingClosedConferencesQuery.cs (hunk 1 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.DAL/Queries/GetExpiredAudiorecordingClosedConferencesQuery.cs]: done

File: VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetExpiredAudiorecordingClosedConferencesQueryTests.cs (hunk 6 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.IntegrationTests/Database/Queries/GetExpiredAudiorecordingClosedConferencesQueryTests.cs]: No, I have 4 conferences and know the Ids, 1st and 4th should be with audio recording and status > 0 and expired, one is closed, second is paused.  So I have checked that my list has 1st and 4th, and not contained 2nd and 3rd.

File: VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs (hunk 4 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs:90]: done

File: VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs (hunk 5 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs:21]: it's a different type and has field HearingId, why it should not be?

File: VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs (hunk 3 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs]: done

File: VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature (hunk 2 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature]: done

File: VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs [hunk 7]
```diff
@@ -278,9 +322,9 @@ private void CreateMultipleConferences(bool addDuplicateFirstNames)
             }
```
Comment (npoxon) [inline VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs:322]: Hi @valentinabennett There is still no assertion that the conferences that you are expecting are retrieved?

File: VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs (hunk 5 above)
Comment (npoxon) [reply VideoAPI/VideoApi.Contract/Responses/ExpiredConferencesResponse.cs:21]: Ok lets leave it like that then.

File: VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature [hunk 8]
```diff
@@ -4,90 +4,100 @@ Feature: Conference
...
+	  And I have a get expired conferences request
+	  When I send the request to the endpoint
+	  Then the response should have the status OK and success status True
+	  And a list not containing the closed hearings should be retrieved
+
+  Scenario: Get audiorecording expired closed conferences
+	  Given I have a conference with audiorecording
+	  And I have another conference with audiorecording
+	  And I have a conference for tomorrow with audiorecording
+	  And All conferences have status greater then zero
```
Comment (npoxon) [inline VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature:90]: What is status greater than zero?

Comment (valentinabennett) [reply VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature:90]: Conference status > 0 ( not started)

Comment (npoxon) [reply VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature:90]: Probably best to change it to All conferences have started

Comment (valentinabennett) [reply VideoAPI/VideoApi.AcceptanceTests/Features/Conference.feature]: Changed to All conferences have started

File: VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs (hunk 7 above)
Comment (valentinabennett) [reply VideoAPI/VideoApi.AcceptanceTests/Steps/ConferenceSteps.cs:324]: The hearing in the past could not be created, so I have check that all known hearings that not meet conditions are not included

**Pull request merged**